library/core/src/num/int_macros.rs

```diff
@@ -2160,6 +2160,11 @@ macro_rules! int_impl {
                       without modifying the original"]
         #[inline(always)]
         #[rustc_allow_const_fn_unstable(unchecked_shifts)]
+        // Precondition: The right hand side of the shift (rhs) is less than
+        // the number of bits in the type. This prevents undefined behavior.
+        // TODO: Is requires needed? It's already specified in `unchecked_shl`
+        #[requires(rhs < Self::BITS)]
```

Comment:
I don't think you need this requires. This function seems to accept any integer.